File: MAP/Lab4/src/Model/Statement/AssignStmt.java
package Model.Statement;

import Model.Expression.Expression;
import Model.Exception.DivByZeroException;
import Model.Exception.EvaluationException;
import Model.IDictionary;
import Model.Exception.InexistentSymbolException;

public class AssignStmt implements Statement {
    private String varName;
    private Expression expr;
    public AssignStmt(String vName, Expression e) {
        varName = vName;
        expr = e;
    }

    @Override
    public PrgState execute(PrgState state) {
        IDictionary<String, Integer> d = state.getSymbolT();
        try
        {
            int intr = expr.Eval(d, state.getHeap());
            if(d.contains(varName))
                d.update(varName, intr);
            else d.add(varName, intr);

        }
        catch(EvaluationException | InexistentSymbolException | DivByZeroException e)
        {
            System.out.println(e.getMessage());
        }
        return state;
    }

    @Override
    public String toString()
    {
        return varName + " = " + expr;
    }
}

File: MAP/LAB7-MAP/LAB7-MAP/Statements/AssignStmt.cs
using System;
using MAP_Lab7.Exceptions;
using MAP_Lab7.Expressions;

namespace MAP_Lab7.Statements
{
    public class AssignStmt:Statement
    {
        private IExpression expr;
        private string var;

        public AssignStmt(string v, IExpression e)
        {
            expr = e;
            var = v;
        }

        public PrgState Execute(PrgState state)
        {
            try
            {

                int result = expr.Eval(state.Dict);
                state.Dict.Update(var, result);

                return state;
            }
            catch(DivByZeroException e)
            {
                Console.WriteLine(e.Message);
            }
            catch(EvaluationException e2)
            {
                Console.WriteLine(e2.Message);
            }
            catch (InexistentSymbolException e3)
            {
                Console.WriteLine(e3.Message);
            }
            return state;
        }

        public override string ToString()
        {
            return var + "=" + expr;
        }
    }
}
